src/main/jbake/content/blog/2017/08/opentracing-appmetrics-canary.adoc

```diff
@@ -0,0 +1,157 @@
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-service-compare.png[]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-service-compare.png[]
+endif::[]
+
+== The updated example application
+
+We will be using the same https://github.com/objectiser/opentracing-prometheus-example[example] as used in the previous
+article.
+
+However since writing that article, the configuration of the tracer and Prometheus metrics support has been
+simplified. There is now no explicit configuration of either, with only some auto configuration of `MetricLabel` beans
+to identify some custom labels to be added to the Prometheus metrics, e.g.
+
+.Metrics configuration used in both services:
+----
+@Configuration
+public class MetricsConfiguration {
+
+    @Bean
+    public MetricLabel transactionLabel() {
+        return new BaggageMetricLabel("transaction", "n/a"); <1>
+    }
+
+    @Bean
+    public MetricLabel versionLabel() {
+        return new ConstMetricLabel("version", System.getenv("VERSION")); <2>
+    }
+
+}
+----
+
+<1> This metric label identifies the business transaction associated with the metrics, which can be used to isolate the
+specific number of requests, duration and errors that occurred when the service was used within the particular business
+transaction
+<2> This metric label identifies the service version, which is especially useful in the Canary deployment use case being
+discussed in this article
+
+The first step is to following the instructions in the https://github.com/objectiser/opentracing-prometheus-example[example]
+for deploying and using the services within OpenShift.
+
+Once the `./genorders.sh` script has been running for a while, to generate plenty of metrics for version `0.0.1` of the
+services, then deploy the new version of the services. This is achieved by:
+
+* updating the versions in the `pom.xml` files, within the `simple/accountmgr` and `simple/ordermgr` folders
+from `0.0.1` to `0.0.2`
+* re-run the `mvn clean install docker:build` command from the `simple` folder
+* deploy the canary versions of the services using the command `oc create -f services-canary-kubernetes.yml`
+
+As our services `accountmgr` and `ordermgr` determine the backing deployment based on the respective labels
+`app: accountmgr` and `app: ordermgr`, simply having a second deployment with these labels will make them serve requests
+in a round-robin manner.
+
+This deployment script has been pre-configured with the `0.0.2` version, and to only start a single instance of the
+new version of the services. This may be desirable if wanting to monitor the behaviour of the new service versions over
+a reasonable time period, but as we want to see results faster we will scale them up to see more activity. This is
+achieved by expanding the deployment area for each service in the OpenShift console, and selecting the up arrow to
```

Comment:
Passive voice, revise to something like "You can do this by expanding the deployment area for each service in the OpenShift web console and selecting the up arrow to scale up each service.